flutter_kernel_transformers/lib/track_widget_constructor_locations.dart

```diff
@@ -456,22 +441,17 @@ class WidgetCreatorTracker implements ProgramTransformer {
       return;
     }
 
-    // TODO(jacobr): once there is a working incremental ClassHierarchy
-    // constructor switch to using it instead of building a ClassHierarchy off
-    // the full program.
-    hierarchy = ClassHierarchy(
-      _computeFullProgram(program),
-      onAmbiguousSupertypes: (Class cls, Supertype a, Supertype b) { },
-    );
-
     final Set<Class> transformedClasses = Set<Class>.identity();
-    final Set<Library> librariesToTransform = Set<Library>.identity()
-      ..addAll(libraries);
+    final Set<Library> librariesToTransform = Set<Library>.identity();
 
     for (Library library in libraries) {
-      if (library.isExternal) {
-        continue;
+      if (!library.isExternal && !_isTransformed(library)) {
+        librariesToTransform.add(library);
```

Comment:
I am wondering if this check is necessary in the [newest SDK repo version](https://dart-review.googlesource.com/c/sdk/+/101987/6/pkg/kernel/lib/transformations/track_widget_constructor_locations.dart) of the transformer, which is given an explicit list of libraries to transform. For this reason, I have not brought the optimization over just yet.

Reply:
Sounds good. Hopefully it isn't needed.